lib/Runtime/PlatformAgnostic/Platform/Linux/PerfTrace.cpp

```diff
@@ -0,0 +1,150 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "Common.h"
+#include "Runtime.h"
+#include "ChakraPlatform.h"
+
+#include <signal.h>
+#include <errno.h>
+#include <unistd.h>
+
+using namespace Js;
+
+//
+//  Handle the SIGUSR2 signal by setting a flag
+//  indicating that we should call WritePerfMap later on
+//
+void handle_signal(int signum)
+{
+    PlatformAgnostic::PerfTrace::mapsRequested = true;
+}
+
+namespace PlatformAgnostic
+{
+
+volatile bool PerfTrace::mapsRequested = false;
+  
+//
+// Registers a signal handler for SIGUSR2 
+//
+void PerfTrace::Register()
+{
+    struct sigaction newAction = {0};
+    newAction.sa_flags = SA_RESTART;
+    newAction.sa_handler = handle_signal;
+    sigemptyset(&newAction.sa_mask);
```

Comment:
Yes; by default I thought we would want to include this (so that people can get useful performance analysis) but if an embedder does not want the signal handler, they can recompile with it disabled?